### PR TITLE
feat(button-group): add justifyContent prop, use flex

### PR DESCRIPTION
**Pull request by nkrantz**
https://twilio-productivity.atlassian.net/browse/DS-445

### CODE REVIEW

**Review by serifluous — APPROVED** [review]

File: packages/paste-core/components/button-group/src/ButtonGroup.tsx [hunk 1]
```diff
@@ -65,11 +63,19 @@ export interface ButtonGroupProps extends HTMLPasteProps<"div"> {
...
    */
   attached?: boolean;
+  /**
+   * Sets the `justify-content` CSS property. Only used for unattached button groups.
+   *
+   * @default "flex-start"
+   * @type {JustifyOptions}
+   * @memberof ButtonGroupProps
+   */
+  justifyContent?: JustifyOptions;
```
Comment (krisantrobus) [inline packages/paste-core/components/button-group/src/ButtonGroup.tsx:73]: question, do we need to expose this prop? For the justify storybook would we really want users to be able to use the display between. We could just update the default value and use the attached boolean to determine whether is need sot be different for the 2 supported variants

Comment (nkrantz) [reply packages/paste-core/components/button-group/src/ButtonGroup.tsx:73]: hmm, you mean making unattached always centered and attached always flex-start? that would be a breaking design change. maybe I'm misunderstanding

Comment (krisantrobus) [reply packages/paste-core/components/button-group/src/ButtonGroup.tsx:73]: I mean leave the styling to how it currently is without any change when attached an then only apply the new flex when they are unattached. No new props for consumers

**Review by krisantrobus — APPROVED** [review]
Happy to move forward, looks great!

**Pull request merged**